fastcrypto-zkp/examples/blake2s_demo.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use ark_bls12_381::{Bls12_381, Fr};
+use ark_ff::ToConstraintField;
+use ark_groth16::{create_random_proof, generate_random_parameters};
+
+use ark_crypto_primitives::{
+    prf::blake2s::constraints::Blake2sGadget, prf::blake2s::Blake2s as B2SPRF, PRFGadget, PRF,
+};
+use ark_relations::r1cs::{
+    ConstraintSynthesizer, ConstraintSystem, ConstraintSystemRef, SynthesisError,
+};
+use blake2::{digest::Digest, Blake2s256};
+
+use ark_r1cs_std::prelude::*;
+use fastcrypto_zkp::verifier::{process_vk_special, verify_with_processed_vk};
+
+#[derive(Clone, Copy, Debug)]
+struct Blake2sCircuit {
```

Comment:
I suggest that you add a short public description doc per our discussion. Otherwise the reader has to read through the inline comments to understand the logic. Pretty much what we discussed. 
```
/// A prover proves pre-image knowledge of a hash output, where the hash input is a <public_seed, secret_string>, and they create a circuit where the witness is the secret_string and the public inputs are the public_seed and the hash_output.
```